roles/core/dns_server/templates/named.conf.j2

```diff
@@ -1,13 +1,24 @@
+#jinja2: lstrip_blocks: True
 #### Blue Banquise file ####
 ## {{ansible_managed}}
 
 options {
   listen-on port 53 {
     127.0.0.1;
 {% for network in (networks | select('match',(j2_current_iceberg_network+'-[0-9]+')) | list | unique | sort) %}
-{% if networks[network]['is_in_dns'] is defined and networks[network]['is_in_dns'] == true %}
-    {{networks[network]['services_ip']['dns_ip']}};
-{% endif %}
+  {% if networks[network]['is_in_dns'] is defined and networks[network]['is_in_dns'] == true %}
+    {% set addresses = networks[network]['services_ip']['dns_ip'] | ipaddr %}
+    {% if dns_slaves is defined %}
```

Comment:
There are no register to define dns_slaves according to `networks[j2_node_main_network].services_ip.dns_ip`, i.e. not first ones. If I understand, first in list is master (if note defined manually), and others are slaves. Is it expected ?

Reply:
dns_slaves can be defined in the general_settings/network.yml inventory file.
This is the list of IP addresses of dns slaves connected to the main network of the master.

  dns_slaves:
    - 10.10.0.2
    - 10.10.0.3

This ensures the master can notify the slaves when there is an update in a DNS map. If dnf_slaves is not defined, the dns master won't know which nodes it has to notify, which will delay the update of the maps on the slaves.

This is useful in multi icebergs where the main network of the master can be ice1-1 while the main network of some slaves is ice2-1. Nodes in ice2-1 will be able to use `networks[j2_node_main_network].services_ip.dns_ip` to request the dns servers (slaves), but the master is not connected to this network.

Reply:
>  If I understand, first in list is master (if note defined manually), and others are slaves. Is it expected ?

Indeed, if `dns_master` is not defined, first element of the list `networks[j2_node_main_network].services_ip.dns_ip` is the master when j2_node_main_network is the primary management network (management_network_naming + '1-1'). Other elements in the list are slaves dns servers available in this network. They are used to render resolv.conf and dhcp configuration (nameservers).

In multi icebergs, an inventory looks like:
```
  inventory:
    vars:
      networks:
        ice1-1:
          subnet: 10.11.0.0
          prefix: 16
          netmask: 255.255.0.0
          broadcast: 10.11.255.255
          is_in_dns: true
          services_ip:
            dns_ip:
              - 10.11.0.1
              - 10.11.0.2
        ice2-1:
          subnet: 10.12.0.0
          prefix: 16
          netmask: 255.255.0.0
          broadcast: 10.12.255.255
          is_in_dns: true
          services_ip:
            dns_ip:
              - 10.12.0.1
              - 10.12.0.2
      external_dns: []
      management_network_naming: "ice"
      domain_name: example.com
      dns_slaves:
        - 10.11.0.2
        - 10.11.0.3
        - 10.11.0.4
      start_services: true
      enable_services: true
    host_vars:
      mgmt0-master:
        j2_current_iceberg_network: ice1
        j2_node_main_network: "ice1-1"
        j2_node_main_network_interface: en0
        network_interfaces:
          en0:
            ip4: 10.11.0.1
            network: ice1-1
      mgmt1-slave:
        j2_current_iceberg_network: ice1
        j2_node_main_network: "ice1-1"
        j2_node_main_network_interface: en0
        network_interfaces:
          en0:
            ip4: 10.11.0.2
            network: ice1-1
      mgmt2-slave:
        j2_current_iceberg_network: ice2
        j2_node_main_network: "ice2-1"
        j2_node_main_network_interface: en1
        network_interfaces:
          en0:
            ip4: 10.11.0.3
            network: ice1-1
          en1:
            ip4: 10.12.0.1
            network: ice2-1
      mgmt3-slave:
        j2_current_iceberg_network: ice2
        j2_node_main_network: "ice2-1"
        j2_node_main_network_interface: en1
        network_interfaces:
          en0:
            ip4: 10.11.0.4
            network: ice1-1
          en1:
            ip4: 10.12.0.2
            network: ice2-1
```

mgmt0-master would be the master DNS server handling the zone files for domain example.com. All other servers will be slaves.
Hosts in network ice1-1 will use nameservers 10.11.0.1 and 10.11.0.2 (`networks["ice1-1"].services_ip.dns_ip`), while hosts in network ice2-1 will use nameservers 10.12.0.1 and 10.12.0.2 (`networks["ice2-1"].services_ip.dns_ip`).

For any update in the DNS map file, the master will notify dns_slaves (10.11.0.[2-4]). The slaves will then start a zone transfer (AXFR) to update their local copy.

Reply:
While writing the code, I also thought using `networks[j2_node_main_network].services_ip.dns_master` and `networks[j2_node_main_network].services_ip.dns_slaves` instead of global dns_master and dns_slaves.
However I find it more error-prone because it might led to think the stack can handle several master/slaves configuration.

Reply:
`if ( dns_slaves is not defined ) and ( nameservers is not string )` -> dns_slave is ignored ?

to confirm, not a DNS master

Reply:
If `dns_slaves is not defined` but `nameservers is a list`, the dns slaves won't be ignored. However, the dns master won't know which IP addresses to notify in network1-1, and the slaves in networkX-1 (X>1) won't listen on the interface connected to network1-1. The slaves will refresh their configuration every 3600 seconds (based on refresh value in the SOA), or when the admin will run the `rndc reload $domain` command.
The `dns_slaves` is needed for master->slaves communication only.

Reply:
Ok for me. Thanks for explaining all of this :-)